src/declarations/types.d.ts

```diff
@@ -0,0 +1,28 @@
+import interfaces from "./interfaces";
+
+export declare type Step = {
+  message: string;
+  action: interfaces.StepAction;
+};
+
+export declare type ValidationResult = {
+  validForNewPackages: boolean;
+  validForOldPackages: boolean;
+  errors?: string[];
+  warnings?: string[];
+};
+
+export declare type Package = {
```

Comment:
there must be a `@types` for package...

https://www.npmjs.com/package/@types/package-json this seems to redirect to a different project but maybe it exists?